tests/components/modbus/test_sensor.py

```diff
@@ -379,7 +379,7 @@ async def test_config_wrong_struct_sensor(
             },
             [9],
             False,
-            "18.5",
+            "18",
```

Comment:
Why is it needed to change this?

Reply:
Because now it return an integer which is rounded, and not a float.

Please see the text in PR, where it is explained.